lib/paperclip/storage/fog.rb

```diff
@@ -141,17 +152,18 @@ def public_url(style = default_style)
 
       def expiring_url(time = (Time.now + 3600), style_name = default_style)
         time = convert_time(time)
-        if path(style_name) && directory.files.respond_to?(:get_http_url)
-          expiring_url = directory.files.get_http_url(path(style_name), time)
-
+        url_method = url_protocol == 'https' ? :get_https_url : :get_http_url
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.